docs/reference/js/api.md

```diff
@@ -5,38 +5,129 @@ description: Developer resources for working in JavaScript with IPFS, the InterP
+:::
 
-## JS API reference
+### HTTP client
 
-See the [JS core API reference →](https://github.com/ipfs/js-ipfs/tree/master/docs/core-api)
+[JS-IPFS HTTP RPC API](https://www.npmjs.com/package/ipfs-http-client) is a client library that controls an active IPFS node (Kubo or JS-IPFS) running through its [RPC API](../kubo/rpc.md).
+  - JS-IPFS will internally use this library if it detects another node is running on your machine. You can also interact with the [RPC API](../kubo/rpc.md) directly using `fetch()` in a browser or a module like `request` in Node.js, but using this library can be much more convenient.
+  - When using JS-IPFS as a backend, use the [ipfs-client](https://www.npmjs.com/package/ipfs-client) instead to leverage gRPC connections over WebSockets to allow some commands to achieve the bidirectional streaming necessary to have full duplex streams running in the browser.
+
+All the libraries have the [same interface](https://github.com/ipfs/js-ipfs/tree/master/docs/core-api) for using all the major IPFS commands. This client library implements the interface-ipfs-core enabling applications to change between an embedded js-ipfs node and any remote IPFS node without changing the code. In addition, this client library implements a set of utility functions.
 
-## RPC client library
+::: tip Interacting with IPFS
 
-A [client library](https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client) for the JS-IPFS RPC API, implemented in JavaScript. This client library implements the interface-ipfs-core enabling applications to change between an embedded js-ipfs node and any remote IPFS node without having to change the code. In addition, this client library implements a set of utility functions.
+We recommend the second method (interacting with a separate IPFS node via RPC API) whenever reasonable. Keeping the IPFS node in a separate process (even if it’s one of your program spawns) isolates you from any stability problems with the node. If a user already has IPFS installed, this also means that you can take advantage of a single, standard installation on their machine. It’s also less code to load in a browser. If you need to spawn a separate IPFS process, you might want to take a look at [`js-ipfsd-ctl`](https://github.com/ipfs/js-ipfsd-ctl), which uses the same interface to spawn a Kubo (Go-IPFS) node, a JS-IPFS node, or an in-process JS-IPFS node.
+
+:::
+
+::: warning Browsers
+
+The `js-ipfs` and `js-ipfs-http-client` libraries work in browsers, but each has special considerations noted in their READMEs.
+
+:::
+
+## Packages
+
+> This table is generated using the module [`package-table`](https://www.npmjs.com/package/package-table) with `package-table --data=package-list.json`.
+
+Listing of the main JS packages in the IPFS ecosystem:
+
+| Package | Version | Deps | CI/Travis | Coverage | Lead Maintainer |
```

Comment:
We're no longer using Travis. I'd remove this column or replace it with GitHub actions badges, e.g. https://github.com/ipfs/js-ipns/blob/master/README.md?plain=1#L7